src/originalhtml.rs

```diff
@@ -0,0 +1,615 @@
+///! # Get original mime-message as HTML.
+///!
+///! Use is_mime_modified() to check if the UI shall render a
+///! corresponding button and get_original_mime_html() to get the full message.
+///!
+///! Even whem the original mime-message is not HTML,
+///! get_original_mime_html() will return HTML -
+///! this allows nice quoting, handling linebreaks properly etc.
+
+use std::future::Future;
+use std::pin::Pin;
+
+use lettre_email::mime::{self, Mime};
+
+use crate::context::Context;
+use crate::error::Result;
+use crate::headerdef::{HeaderDef, HeaderDefMap};
+use crate::message::{Message, MsgId};
+use crate::mimeparser::parse_message_id;
+use crate::simplify::split_lines;
+use mailparse::ParsedContentType;
+use once_cell::sync::Lazy;
+
+impl Message {
+    /// Check if the mime structure of a Message is modified.
+    /// "Modified" here means that some text is cut or the original message
+    /// is in HTML and simplify() may hide some maybe important information.
+    /// The corresponding ffi-function is dc_msg_is_mime_modified().
+    pub fn is_mime_modified(&self) -> bool {
+        self.mime_modified
+    }
+}
+
+/// Type defining a rough mime-type.
+/// This is mainly useful on interating
+/// to decide whether a mime-part has subtypes.
+enum MimeS {
```

Comment:
What does "S" mean here?

Reply:
i just reused that from mimeparser.rs, i am also wondering. i read that as "super" :)

Reply:
i'll rename that to `enum MimeMultipartType` i think, this is the gist of the thing. at least, it is better than `MimeS` :)